src/consumerIncentives/saga.ts

```diff
@@ -126,18 +150,78 @@ function* claimReward(reward: SuperchargePendingReward, index: number, baseNonce
   }
 }
 
+function* claimRewardV2(reward: SuperchargePendingRewardV2, index: number, baseNonce: number) {
+  const { transaction, details } = reward
+
+  const superchargeRewardContractAddress = yield select(superchargeRewardContractAddressSelector)
+  if (superchargeRewardContractAddress !== transaction.to) {
+    throw new Error(
+      `Unexpected supercharge contract address ${transaction.to} on reward transaction, aborting claim.`
+    )
+  }
+
+  const kit: ContractKit = yield call(getContractKit)
+  const tokens: TokenBalances = yield select(tokensByAddressSelector)
+  const walletAddress: string = yield call(getConnectedUnlockedAccount)
+
+  Logger.debug(TAG, `Start claiming reward at index ${index}: ${JSON.stringify(reward)}`)
+
+  applyChainIdWorkaround(transaction, yield call([kit.connection, 'chainId']))
```

Comment:
nit: I think this is not needed... I tested locally and I was able to send the transaction as it is returned by the API.

Reply:
hmmmm maybe...i don't fully understand the scenarios where this workaround makes a differentce but there is a [long comment](https://github.com/valora-inc/wallet/blob/9ccce4df6a9ad4e836118e5961028b83942dc75d/src/web3/utils.ts#L64-L69) mentioning a `bug in TxParamsNormalizer` so i'm reluctant to remove it. it's used in other places in the same way, and it doesn't seem to cause issues (and it does do something, the `chainId` without this workaround is a number but after this workaround is a `0x` base 16 string)

Reply:
`applyChainIdWorkaround` is really needed to sign WC transactions where the `chainId` is stripped by WC.
But it should be harmless to keep it.

Reply:
ah thanks for the info! i didn't know, i removed it since it seems it's not needed